File: src-tauri/src/models/bot_difficulty.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BotProfileSummary {
    pub id: String,
    pub name: String,
    pub source: String,
    pub base_difficulty: Option<String>,
    pub active: bool,
    pub read_only: bool,
    pub db_sha256: Option<String>,
    pub vpk_sha256: Option<String>,
    pub warnings: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BotProfileList {
    pub profiles: Vec<BotProfileSummary>,
    pub tool_version: Option<String>,
    pub tool_sha256: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BotProfileDocument {
    pub profile: BotProfileSummary,
    pub text: Option<String>,
    pub entry_path: String,
    pub validation: String,
    pub dirty: bool,
    pub tool_version: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateBotProfileRequest {
    pub base_profile_id: String,
    pub name: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SaveBotProfileRequest {
    pub profile_id: String,
    pub text: String,
    pub expected_db_sha256: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BotProfileOperation {
    pub profile: BotProfileSummary,
    pub backup_path: Option<String>,
    pub applied: bool,
    pub message: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BotToolState {
    pub status: String,
    pub version: String,
    pub sha256: Option<String>,
    pub path_hint: Option<String>,
    pub detail: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct VpkEntry {
    pub path: String,
    pub size: u64,
    pub sha256: Option<String>,
}

File: src/services/tauri/bot-difficulty.ts
import { invoke } from '@tauri-apps/api/core'

export interface BotProfileSummary { id: string; name: string; source: 'builtin' | 'custom'; baseDifficulty?: string; active: boolean; readOnly: boolean; dbSha256?: string; vpkSha256?: string; warnings: string[] }
export interface BotProfileList { profiles: BotProfileSummary[]; toolVersion?: string; toolSha256?: string }
export interface BotProfileDocument { profile: BotProfileSummary; text?: string; entryPath: string; validation: string; dirty: boolean; toolVersion?: string }
export interface BotToolState { status: 'missing' | 'invalid' | 'ready'; version: string; sha256?: string; pathHint?: string; detail?: string }
export interface BotProfileOperation { profile: BotProfileSummary; backupPath?: string; applied: boolean; message: string }
export interface CreateBotProfileRequest { baseProfileId: string; name: string }
export interface SaveBotProfileRequest { profileId: string; text: string; expectedDbSha256: string }
export interface RenameBotProfileRequest { profileId: string; name: string }
export const listBotProfiles = (rootPath: string) => invoke<BotProfileList>('list_bot_profiles', { rootPath })
export const openBotProfile = (rootPath: string, profileId: string) => invoke<BotProfileDocument>('open_bot_profile', { rootPath, profileId })
export const getBotWorkshopState = () => invoke<BotToolState>('get_bot_workshop_state')
export const createBotProfile = (rootPath: string, request: CreateBotProfileRequest) => invoke<BotProfileOperation>('create_bot_profile', { rootPath, request })
export const saveBotProfile = (rootPath: string, request: SaveBotProfileRequest) => invoke<BotProfileOperation>('save_bot_profile', { rootPath, request })
export const renameBotProfile = (request: RenameBotProfileRequest) => invoke<BotProfileOperation>('rename_bot_profile', { request })
export const deleteBotProfile = (rootPath: string, profileId: string) => invoke<BotProfileOperation>('delete_bot_profile', { rootPath, profileId })
export const applyBotProfile = (rootPath: string, profileId: string) => invoke<BotProfileOperation>('apply_bot_profile', { rootPath, profileId })
